cpp/src/arrow/array/util.h

```diff
@@ -54,6 +54,17 @@ ARROW_EXPORT
 Result<std::shared_ptr<Array>> MakeArrayFromScalar(
     const Scalar& scalar, int64_t length, MemoryPool* pool = default_memory_pool());
 
+/// \brief Create an empty Array of a given type
+///
+/// The output Array will be of the given type.
+///
+/// \param[in] type DataType of the empty Array
+/// \param[in] pool MemoryPool pointer
```

Comment:
This is not really informative. Make this for example "the memory pool to allocate from", like above?